src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/branch/BitbucketCloudBranch.java

```diff
@@ -23,72 +23,64 @@
  */
 package com.cloudbees.jenkins.plugins.bitbucket.client.branch;
 
-import org.codehaus.jackson.annotate.JsonIgnoreProperties;
-import org.codehaus.jackson.annotate.JsonProperty;
-
 import com.cloudbees.jenkins.plugins.bitbucket.api.BitbucketBranch;
+import com.cloudbees.jenkins.plugins.bitbucket.client.repository.BitbucketCloudRepository;
+import org.codehaus.jackson.annotate.JsonCreator;
+import org.codehaus.jackson.annotate.JsonProperty;
 
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
-@JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
Why is this removed? It will fail the deserilization if some field is not coming in the JSON response. Do we want that? In general I think it is better to do null checks when needed and fail gracefully if needed.

Reply:
`BitbucketCloudApiClient` now uses single instance of ObjectMapper configured to not fail on unknown properties. Using single instance of ObjectMapper is recommended, avoids repeating such annotations everywhere, better for performance  and is thread safe so no issue.

See:
https://github.com/vivek/bitbucket-branch-source-plugin/blob/49e2d0791848ded3f5281279ef73bf7e7ebdf84b/src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java#L747

Reply:
Ok, then all `@JsonIgnoreProperties` should be removed to be consistent.

Reply:
Yes, already done, will push after some testing